cmd/nerdctl/network_create.go

```diff
@@ -96,34 +97,18 @@ func networkCreateAction(cmd *cobra.Command, args []string) error {
 	}
 	labels = strutil.DedupeStrSlice(labels)
 
-	if subnetStr == "" {
-		if gatewayStr != "" || ipRangeStr != "" {
-			return fmt.Errorf("cannot set gateway or ip-range without subnet, specify --subnet manually")
-		}
-	}
-
-	e, err := netutil.NewCNIEnv(globalOptions.CNIPath, globalOptions.CNINetConfPath)
-	if err != nil {
-		return err
-	}
-	createOpts := netutil.CreateOptions{
-		Name:        name,
-		Driver:      driver,
-		Options:     strutil.ConvertKVStringsToMap(opts),
-		IPAMDriver:  ipamDriver,
-		IPAMOptions: strutil.ConvertKVStringsToMap(ipamOpts),
-		Subnet:      subnetStr,
-		Gateway:     gatewayStr,
-		IPRange:     ipRangeStr,
-		Labels:      labels,
-	}
-	net, err := e.CreateNetwork(createOpts)
-	if err != nil {
-		if errdefs.IsAlreadyExists(err) {
-			return fmt.Errorf("network with name %s already exists", name)
-		}
-		return err
-	}
-	_, err = fmt.Fprintf(cmd.OutOrStdout(), "%s\n", *net.NerdctlID)
-	return err
+	return network.Create(&types.NetworkCreateCommandOptions{
```

Comment:
Can we change the `options` to non pointers? It's safer than pointers (in case they're shared by more than 1 goroutines) and avoid heap allocation/gc